### ISSUE TITLE
Website SSL issue

### ISSUE THREAD

**Issue by sebastianharko** [reactivemanifesto/reactivemanifesto#107]
Getting an SSL related error in Chrome when opening the website.

See:
https://www.sslshopper.com/ssl-checker.html#hostname=https://www.reactivemanifesto.org

_"None of the common names in the certificate match the name that was entered (www.reactivemanifesto.org). You may receive an error when accessing this site in a web browser"_

:-)

**Comment by jroper** [comment]
Yes, we haven't enabled https for the site yet, there's a non trivial amount of work we need to do involving social logins and also mixed content issues before that can be done.

Was there an https link somewhere that you clicked on to get to the site? If so, we should get that updated because it's the wrong link, all links should currently be http.

**Comment by rkuhn** [comment]
It was in a tweet, so I don't think it is reasonable to try and control what somebody types into a message text field. The time might have come when https should be considered mandatory.

**Comment by jroper** [comment]
Get one character wrong in a URL, and the URL is wrong. Out of the billions of possibilities of likely mistakes that a user could possibly make when typing a URL, only a very rare few, such as this one, can we do something about. So yes, it is reasonable to place the burden of getting URLs right on users, just as for any addressing system the burden of getting the address right is on the sender, and in the case of URLs they can do that easily by using copy/paste of an actual working URL (I don't know about you but I almost never hand type URLs into text fields, I will hand type them into my browser, confirm that the page works and that the URL I have is the canonical URL, then copy/paste it).

**Comment by rkuhn** [comment]
James, the tweets (another one yesterday) that I meant are from Oliver and Kat. It might just be that https:// is assumed by Tweetdeck, I don’t know, but my point is that this particular “mistake” is not considered nor expected by internet users today. If HTTPS is not supported, then the server should at least not respond on that port.

**Comment by jroper** [comment]
Not much we can do about the server not responding on https port, it's the cloud, the server accepting https connections is a load balancer shared by thousands of other websites that we have little control over.

Anyway I spent all day yesterday working on this, including migrating the existing dataset to use https urls for avatars, auditing the APIs to ensure new signatures would only ever get https avatar URLs, and recreating new social apps for all the oauth signins because no one knows who owns the old social app accounts to reconfigure them, and now we have https mostly working - GitHub signin doesn't work when using https because it doesn't support configuring both http and https redirect URLs, if I changed it to https that would break http users, where the majority of the web comes from. I have to write a filter to upgrade all requests to https.
